SConstruct

```diff
@@ -700,3 +700,16 @@ if "env" in locals():
     # TODO: replace this with `env.Dump(format="json")`
     # once we start requiring SCons 4.0 as min version.
     methods.dump(env)
+
+    import atexit
+
+    def check_build_failures():
+        failed = len(GetBuildFailures()) > 0
+        if failed and env["werror"] and not env.stable_release:
```

Comment:
It shouldn't be specific to development version, disabling `-Werror` to bypass warnings is always a valid workaround. The only difference on stable releases is that it's disabled already, but if users enable it manually and get errors, they might appreciate the hint too.

Reply:
I agree, if a user passes `dev=yes` then it might not be obvious what specific `werror` value is set.